core/src/main/java/hudson/util/RingBufferLogHandler.java

```diff
@@ -86,13 +75,36 @@ public static int getDefaultRingBufferSize() {
     }
 
     @Override
-    public synchronized void publish(LogRecord record) {
-        int len = records.length;
-        records[(start + size) % len] = new LogRecordRef(record);
-        if (size == len) {
-            start = (start + 1) % len;
-        } else {
-            size++;
+    public void publish(LogRecord record) {
+        if (record == null) {
+            return;
+        }
```

Comment:
A call to `super.isLoggable(record)` does this null check, and it also checks whether the `LogRecord` has an appropriate level and whether it satisfies any filter. In practice, the level check is handled by `hudson.logging.LogRecorder.Target#matches` and subclasses are unlikely to implement a filter, but calling `super.isLoggable(record)` here results in a slight deduplication of code and makes this subclass of `Handler` consistent with the subclasses of `Handler` provided by the Java Platform, so I think this is worth doing.

```suggestion
        if (!super.isLoggable(record)) {
            return;
        }
```

Reply:
I will have to double-check, but I think this is what we do _not_ want to do because this would prevent custom fine loggers from getting anything. Regular handlers either print a message to a sink or do not; handlers which collect all messages need to behave differently. https://github.com/jenkinsci/support-core-plugin/blob/7e75fdc7e9af0abe0d7d659a3e6e591c3451288f/src/main/java/com/cloudbees/jenkins/support/SupportLogHandler.java#L128-L135

Reply:
> This would prevent custom fine loggers from getting anything

I do not see how it could because we never mutate `java.util.logging.Handler#logLevel` from its default value of `ALL` in `RingBufferLogHandler`. Like I wrote earlier, I do not think there is a difference in practice, but I think using the built-in method better adheres to the parent class's extensible design just in case a theoretical subclass of `RingBufferLogHandler` wanted to use a custom level or filter.

Reply:
It appears to work. Still, I would rather not do it here because

* I am not entirely confident it is correct.
* It is not much related to the rest of the change. (I merely added the null check after noticing that `tooMuchRecordsShouldNotCrashHandler` was not testing anything.)

Reply:
> I would rather just make it final than consider such possibilities.

I am OK with making it `final`.